src/t8_schemes/t8_default/t8_default_pyramid/t8_dpyramid.h

```diff
@@ -55,7 +55,17 @@ typedef int32_t     t8_dpyramid_coord_t;
 */
 typedef int8_t t8_dpyramid_type_t;
 
-/** This data type stores a pyramid. */
-typedef t8_dtet_t t8_dpyramid_t;
+/**
+ * This data type stores a pyramid. 
+ * The coordinates, the level and the type of a pyramid are stored in the tet-struct \a pyramid.
+ * Level, at which the shape switches from tet, to pyra. -1 if not computed for a pyramid with the shape of a tet
+ * undefined, if the pyramid has the shape of a pyramid.
+ */
+typedef struct t8_dpyramid
+{
+  t8_dtet_t           pyramid;  /* Coordinates, level and type */
```

Comment:
I'm wondering if this is the most elegant solution in order to extend the tetraeder type to a pyramid type.

I think it is worth to think about if instead we define the fields explicitly:
```
typedef struct t8_dpyramid
{
#ifdef T8_ENABLE_DEBUG
  /* TODO: enum to int8??? #defines ?? */
  int8_t              eclass_int8;
#endif
  /** The refinement level of the tetrahedron relative to the root at level 0. */
  int8_t              level;

  /** Type of the tetrahedron in 0, ..., 5. */
  t8_dtet_type_t      type;

  t8_dtet_coord_t     x;        /**< The x integer coordinate of the anchor node. */
  t8_dtet_coord_t     y;        /**< The y integer coordinate of the anchor node. */
  t8_dtet_coord_t     z;        /**< The z integer coordinate of the anchor node. */

   /**< Pyramid specific fields. */
  int8_t switch_shape_at_level;
}
t8_dpyramid_t;
```

Whenever you are calling tetraeder functions you can do a type casting:
```
do_something_tet_specific((t8_dtet_t *) my_pyramid_ptr);
```
This actually works, since the tetraeder fields are at the beginning of the struct.

If you do it this way, you don't have to change so much in `*_bits.c`.

Reply:
I have mixed feelings about this. Yes, it would works, but that way we give over the controll to the typecasting. In the current way, all controll is in our hands, which I prefer.

Reply:
So, you don't trust your compiler? :P